pydra/tasks/mriqc/workflows/anatomical/base.py

```diff
@@ -982,33 +956,32 @@ def airmsk_wf(name="AirMaskWorkflow"):
         ),
         name="inputnode",
     )
-    outputnode = pe.Node(
-        niu.IdentityInterface(fields=["hat_mask", "air_mask", "art_mask", "rot_mask"]),
+    outputnode = pydra.Node(
+        interface=specs.IdentityInterface(fields=["hat_mask", "air_mask", "art_mask", "rot_mask"]),
         name="outputnode",
     )
 
-    rotmsk = pe.Node(RotationMask(), name="RotationMask")
-    qi1 = pe.Node(ArtifactMask(), name="ArtifactMask")
+    rotmsk = ShellCommandTask(name="RotationMask", executable="RotationMask")
+    qi1 = ShellCommandTask(name="ArtifactMask", executable="ArtifactMask")
 
-    # fmt: off
-    workflow.connect([
-        (inputnode, rotmsk, [("in_file", "in_file")]),
-        (inputnode, qi1, [("in_file", "in_file"),
-                          ("head_mask", "head_mask"),
-                          ("ind2std_xfm", "ind2std_xfm")]),
-        (qi1, outputnode, [("out_hat_msk", "hat_mask"),
-                           ("out_air_msk", "air_mask"),
-                           ("out_art_msk", "art_mask")]),
-        (rotmsk, outputnode, [("out_file", "rot_mask")])
-    ])
-    # fmt: on
+    workflow.add(rotmsk)
+    workflow.add(qi1)
+
+    workflow.inputs += [("inputnode.in_file", inputnode.inputs.in_file)]
```

Comment:
Instead of the connections being set at the workflow level, you set the inputs of the nodes with lzout placeholders

```python
workflow.rotmsk.inputs.in_file = workflow.lzin.in_file
workflow.qi1.inputs.in_file = workflow.lzin.in_file
workflow.qi1.inputs.head_mask = workflow.lzin.head_mask
```

Reply:
If you want to connect and output from an upstream node you can use the `lzout` attribute of that node, e.g.

```python
workflow.qi1.inputs.head_mask = workflow.name_of_upstream_node.lzout.head_mask
```